tests/suites/storage/charm_storage.sh

```diff
@@ -18,105 +16,110 @@ run_charm_storage() {
 	# Assess the above created storage pools.
 	echo "Assessing storage pool"
 	if [ "${BOOTSTRAP_PROVIDER:-}" == "ec2" ]; then
-		juju list-storage-pools -m "${model_name}" --format json | jq '.ebs | .provider' | check "ebs"
-		juju list-storage-pools -m "${model_name}" --format json | jq '.["ebs-ssd"] | .provider' | check "ebs"
-		juju list-storage-pools -m "${model_name}" --format json | jq '.tmpfs | .provider' | check "tmpfs"
-		juju list-storage-pools -m "${model_name}" --format json | jq '.loop | .provider' | check "loop"
-		juju list-storage-pools -m "${model_name}" --format json | jq '.rootfs | .provider' | check "rootfs"
+		juju storage-pools -m "${model_name}" --format json | jq '.ebs | .provider' | check "ebs"
+		juju storage-pools -m "${model_name}" --format json | jq '.["ebs-ssd"] | .provider' | check "ebs"
+		juju storage-pools -m "${model_name}" --format json | jq '.tmpfs | .provider' | check "tmpfs"
+		juju storage-pools -m "${model_name}" --format json | jq '.loop | .provider' | check "loop"
+		juju storage-pools -m "${model_name}" --format json | jq '.rootfs | .provider' | check "rootfs"
 	else
-		juju list-storage-pools -m "${model_name}" --format json | jq '.rooty | .provider' | check "rootfs"
-		juju list-storage-pools -m "${model_name}" --format json | jq '.tempy | .provider' | check "tmpfs"
-		juju list-storage-pools -m "${model_name}" --format json | jq '.loopy | .provider' | check "loop"
-		juju list-storage-pools -m "${model_name}" --format json | jq '.ebsy | .provider' | check "ebs"
+		juju storage-pools -m "${model_name}" --format json | jq '.rooty | .provider' | check "rootfs"
+		juju storage-pools -m "${model_name}" --format json | jq '.tempy | .provider' | check "tmpfs"
+		juju storage-pools -m "${model_name}" --format json | jq '.loopy | .provider' | check "loop"
+		juju storage-pools -m "${model_name}" --format json | jq '.ebsy | .provider' | check "ebs"
 	fi
 	echo "Storage pool PASSED"
 
 	# Assess charm storage with the filesystem storage provider
 	echo "Assessing filesystem rootfs"
 	juju deploy ./testcharms/charms/dummy-storage-fs --series="jammy" --storage data=rootfs,1G
-	wait_for "dummy-storage-fs" ".applications"
+	wait_for "dummy-storage-fs" "$(idle_condition "dummy-storage-fs" 0 0)"
```

Comment:
0 is the default param for the app index and unit index, so this can be simplified here and elsewhere